src/test/java/uk/gov/hmcts/sptribs/caseworker/event/page/ManageSelectOrdersTest.java

```diff
@@ -19,43 +21,72 @@
 import java.util.UUID;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
 
 @ExtendWith(MockitoExtension.class)
 public class ManageSelectOrdersTest {
 
     @InjectMocks
     private ManageSelectOrders manageSelectOrders;
 
+    @Mock
+    private ListValue<DateModel> mockListValueDateModel;
+
+    private final DynamicList dynamicListLabelled = createDynamicList("0");
+
+    private final DynamicList dynamicListUnlabelled = createDynamicList("");
+
     @Test
-    void shouldReturnErrorsIfNoNotificationPartySelected() {
-        //Given
+    void midEventCompletesSuccessfully() {
         final CaseDetails<CaseData, State> caseDetails = new CaseDetails<>();
         final CaseData caseData = CaseData.builder().build();
         final Order order = Order.builder()
-            .dueDateList(null)
+            .dueDateList(List.of(mockListValueDateModel))
             .draftOrder(new DraftOrderCIC())
             .build();
-        ListValue<Order> listValue = new ListValue<>();
+        final ListValue<Order> listValue = new ListValue<>();
         listValue.setValue(order);
         listValue.setId("0");
         final CicCase cicCase = CicCase.builder()
             .orderList(List.of(listValue))
-            .orderDynamicList(geOrderList())
+            .orderDynamicList(dynamicListLabelled)
             .build();
         caseData.setCicCase(cicCase);
         caseDetails.setData(caseData);
 
-        //When
-        AboutToStartOrSubmitResponse<CaseData, State> response = manageSelectOrders.midEvent(caseDetails, caseDetails);
+        final AboutToStartOrSubmitResponse<CaseData, State> response = manageSelectOrders.midEvent(caseDetails, caseDetails);
+        assertNotNull(cicCase.getOrderDueDates());
+        assertThat(cicCase.getOrderDueDates()).isEqualTo(List.of(mockListValueDateModel));
+        assertThat(response.getErrors()).isEmpty();
+    }
+
+    @Test
+    void midEventReturnsErrorWhenNoSelectedOrder() {
+        final CaseDetails<CaseData, State> caseDetails = new CaseDetails<>();
+        final CaseData caseData = CaseData.builder().build();
+        final Order order = Order.builder()
+            .dueDateList(List.of(mockListValueDateModel))
+            .draftOrder(new DraftOrderCIC())
+            .build();
+        final ListValue<Order> listValue = new ListValue<>();
+        listValue.setValue(order);
+        listValue.setId("0");
+        final CicCase cicCase = CicCase.builder()
+            .orderList(List.of(listValue))
```

Comment:
We may need to add new test cases when the orderlist is empty or null.

Reply:
Added new test cases but had to amend ManageSelectOrders for NPE.